default_warehouse_from_sale_team/models/ir_rule.py

```diff
@@ -0,0 +1,25 @@
+from odoo import _, models
+from odoo.exceptions import AccessError
+
+
+class IrRule(models.Model):
+    _inherit = "ir.rule"
+
+    def _make_access_error(self, operation, records):
+        """Ensure that the selected journal is accessible for the user's team/branch.
+        If no team are defined for the journal, access is open to all users.
+        """
+        if records and records._name == "account.move":
+            journals = records.mapped("journal_id").sudo()
+            if journals:
```

Comment:
This is re-writing the access error message in al,most all cases, as long as the journal entries have a journal, independently of what caused it. You should assure it's caused only by the associated rule by using `self._get_failing(records, mode)`